### PR TITLE
Bug 1885493 - signingscript: add autograph_xpi formats, enable them in adhoc-t

### PR DESCRIPTION
**Pull request by jcristau**

### CODE REVIEW

File: signingscript/docker.d/passwords.yml [hunk 1]
```diff
@@ -200,6 +200,11 @@ in:
              {"$eval": "AUTOGRAPH_GPG_PASSWORD"},
              ["autograph_gpg"]
             ]
+          - ["https://autograph-external.prod.autograph.services.mozaws.net",
```
Comment (JohanLorenzo) [inline signingscript/docker.d/passwords.yml]: I might be wrong. My understanding was that we would talk to the staging instance of autograph. If that's actually the case, we should replace this URL by: 

```suggestion
          - ["https://autograph-external.stage.autograph.services.mozaws.net",
```

Comment (jcristau) [reply signingscript/docker.d/passwords.yml]: Quite right!

**Review by JohanLorenzo — APPROVED** [review]
This looks great to me. Thanks for adding all the tests. I'm confident this is the right approach

File: signingscript/src/signingscript/task.py [hunk 2]
```diff
@@ -160,6 +160,8 @@ async def sign(context, path, signing_formats, **kwargs):
 
 
 def _get_signing_function_from_format(fmt):
+    if fmt.startswith("autograph_xpi"):
```
Comment (willdurand) [inline signingscript/src/signingscript/task.py:163]: for consistency:

```suggestion
    if fmt.startswith("autograph_xpi_"):
```

Comment (jcristau) [reply signingscript/src/signingscript/task.py:163]: That's on purpose, I wanted the bare `autograph_xpi` to just use the default options (SHA256 / ES256).

Comment (willdurand) [reply signingscript/src/signingscript/task.py:163]: ha, sorry about that.

**Pull request merged**